cpp/ql/lib/upgrades/3d35dd6b50edfc540c14c6757e0c7b3c5b7b04dd/usings.ql

```diff
@@ -0,0 +1,17 @@
+class UsingEntry extends @using {
+  string toString() { none() }
+}
+
+class Element extends @element {
+  string toString() { none() }
+}
+
+class Location extends @location_default {
+  string toString() { none() }
+}
+
+from UsingEntry u, Element target, Location loc, int kind
+where
+  usings(u, target, loc) and
+  if target instanceof @namespace then kind = 2 else kind = 1
+select u, target, loc, kind
```

Comment:
I assume you did some basic testing to ensure this works as expected?

Reply:
Yes, I spot checked the version you reviewed by creating a database for the usings test with the before/after CLI and running the queries with after/before CLIs respectively. I'm doing the same for the `using_container` fix now.

Reply:
Just confirmed all looks good with the additional typo fix in https://github.com/github/codeql/pull/17006/commits/3defc8b5de03fb9219b0cacf520d69c1a597604e.